### PR TITLE
Fix "breaking then right-clicking a chest causes a crash" in Minetest Game

### LINKED ISSUE #1462: With no smooth lighting EW faces should be brighter than NS faces
With shaders off and smooth lighting on, EW node faces are brighter than NS faces, this seems best to me since the EW faces face the path of the sun, sunrise and sunset.
With shaders off and smooth lighting off, NS faces are brighter than EW faces, please could this be reversed to be consistent with smooth lighting on?
I seem to remember that Minetest used to have NS faces brighter in all modes, were there good reasons for this?

### PR DESCRIPTION
**Pull request by presstabstart**
Put info in the issue:

"This is definitely an engine bug. It had to do with the fact that a temp variable, meta, was not being set to null after the dig function if the node in question was deleted. As a result, if you finished digging and right clicked in the same frame the pointer named 'meta' was invalidated, effectively crashing the game when calling the 'getString' function of the meta object.

Could somebody please test my fix? I'm about to send it as a PR to the minetest repo, as it's an engine bug. It's a very hard bug to reproduce so I could be wrong.

apologies if I'm wrong and the fix doesn't work."

EDIT by paramat: fixes https://github.com/minetest/minetest_game/issues/1462

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -3825,9 +3825,9 @@ void Game::handlePointingAtNode(GameRunData *runData,
 	}
 
 	if (runData->nodig_delay_timer <= 0.0 && isLeftPressed()
-			&& client->checkPrivilege("interact")) {
-		handleDigging(runData, pointed, nodepos, playeritem_toolcap, dtime);
-	}
+			&& client->checkPrivilege("interact")
+			&& handleDigging(runData, pointed, nodepos, playeritem_toolcap, dtime))
+			meta = NULL; // If node is removed, pointer to metadata no longer valid.
```
Comment (SmallJoker) [inline src/game.cpp]: One indent too much for `meta = NULL`.

Comment (presstabstart) [inline src/game.cpp]: OK, fixed now

**Comment by SmallJoker** [comment]
Needs rebase.

**Comment by nerzhul** [comment]
hmmm but we don't have CSM and meta client side at this moment, the function you modify is client side and the issue is server side. This doesn't fix the problem

**Comment by paramat** [comment]
Bug needs retesting due to new chests.

**Comment by Fixer-007** [comment]
Since bug is still present, why this PR still sitting here 3 months?

**Comment by paramat** [comment]
Bug present with new chests.

**Comment by paramat** [comment]
@presstabstart this may or may not need rework due to new chests.

**Comment by paramat** [comment]
Due to nerzhul's comment above this should be tested and reconsidered.

**Comment by paramat** [comment]
@Fixer-007 does this PR fix the bug?

**Comment by Fixer-007** [comment]
Please rebase, so I can test

**Comment by paramat** [comment]
Looking at the author's activity, they may be absent for a while. Perhaps best if someone rebases this and opens a new PR?

**Comment by paramat** [comment]
But then, i'm concerned about nerzhul's comment.
Is this bug really so hard to trigger?

**Comment by paramat** [comment]
I'm rebasing this to open in a new PR.

**Comment by paramat** [comment]
Replaced by #5801

**Pull request closed without merging**